new pip pre failures

```
______________________________ test_base_epochs _______________________________
mne\tests\test_epochs.py:709: in test_base_epochs
    BaseEpochs(raw.info, None, (np.ones((1, 3), int), {'foo': 1}))
<decorator-gen-247>:12: in __init__
    ???
mne\epochs.py:408: in __init__
    events = _ensure_events(events)
mne\utils\check.py:1002: in _ensure_events
    events = np.asarray(events)
E   ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
______________________________ test_events_type _______________________________
mne\tests\test_epochs.py:807: in test_events_type
    Epochs(raw, events, event_id, tmin, tmax)
<decorator-gen-259>:12: in __init__
    ???
mne\epochs.py:2590: in __init__
    super(Epochs, self).__init__(
<decorator-gen-247>:12: in __init__
    ???
mne\epochs.py:408: in __init__
    events = _ensure_events(events)
mne\utils\check.py:1002: in _ensure_events
    events = np.asarray(events)
E   ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
```

probably newly an error in numpy 1.24.dev0

--- a/mne/utils/check.py
+++ b/mne/utils/check.py
@@ -996,13 +996,26 @@ def _safe_input(msg, *, alt=None, use=None):
 
 
 def _ensure_events(events):
-    events_type = type(events)
-    with warnings.catch_warnings(record=True):
-        warnings.simplefilter('ignore')  # deprecation for object array
-        events = np.asarray(events)
+    err_msg = f'events should be a NumPy array of integers, got {type(events)}'
+    if check_version('numpy', '1.24'):
+        # NumPy >= 1.24 raises ValueError
+        try:
+            events = np.asarray(events)
+        except ValueError as np_err:
+            if np_err.startswith(
+                    'setting an array element with a sequence. The requested '
+                    'array has an inhomogeneous shape'):
+                raise TypeError(err_msg)
+            else:
+                raise
+    else:
+        # NumPy < 1.24 gives VisibleDeprecationWarning
+        with warnings.catch_warnings(record=True):
+            warnings.simplefilter('ignore')
+            events = np.asarray(events)
+    # now that we're sure that events is an array...
     if not np.issubdtype(events.dtype, np.integer):
-        raise TypeError('events should be a NumPy array of integers, '
-                        f'got {events_type}')
+        raise TypeError(err_msg)
     if events.ndim != 2 or events.shape[1] != 3:
         raise ValueError(
             f'events must be of shape (N, 3), got {events.shape}')

--- a/mne/utils/check.py
+++ b/mne/utils/check.py
@@ -1002,7 +1002,7 @@ def _ensure_events(events):
         try:
             events = np.asarray(events)
         except ValueError as np_err:
-            if np_err.startswith(
+            if str(np_err).startswith(
                     'setting an array element with a sequence. The requested '
                     'array has an inhomogeneous shape'):
                 raise TypeError(err_msg)

